pysteps/blending/steps.py

```diff
@@ -0,0 +1,1746 @@
+
+    # Check if NWP models/members should be used individually, or if all of
+    # them are blended together per nowcast ensemble member.
+    if blend_nwp_members == True:
+        # stack the extrapolation cascades into a list containing all ensemble members
+        R_c = [
+            [R_c[j].copy() for j in range(n_cascade_levels)]
+            for i in range(n_ens_members)
+        ]
+        R_c = np.stack(R_c)
+
+    else:
+        # Start with determining the maximum and mimimum number of members/models
+        # in both input products
+        n_model_members = R_models.shape[0]
+        n_ens_members_max = max(n_ens_members, n_model_members)
+        n_ens_members_min = min(n_ens_members, n_model_members)
+
+        # Now, repeat the nowcast ensemble members or the nwp models/members until
+        # it has the same amount of members as n_ens_members_max. For instance, if
+        # you have 10 ensemble nowcasts members and 3 NWP members, the output will
+        # be an ensemble of 10 members. Hence, the three NWP members are blended
+        # with the first three members of the nowcast (member one with member one,
+        # two with two, etc.), subsequently, the same NWP members are blended with
+        # the next three members (NWP member one with member 4, NWP member 2 with
+        # member 5, etc.), until 10 is reached.
+        if n_ens_members_min != n_ens_members_max:
+            if n_model_members == 1:
+                R_models = np.repeat(R_models[:, :, :, :, :], n_ens_members_max, axis=0)
+                mu_models = np.repeat(mu_models[:, :, :], n_ens_members_max, axis=0)
+                sigma_models = np.repeat(
+                    sigma_models[:, :, :], n_ens_members_max, axis=0
+                )
+                V_models = np.repeat(V_models[:, :, :, :], n_ens_members_max, axis=0)
+                # For the prob. matching
+                R_models_pm = np.repeat(
+                    R_models_pm[:, :, :, :], n_ens_members_max, axis=0
+                )
+
+            elif n_model_members == n_ens_members_min:
+                repeats = [
+                    (n_ens_members_max + i) // n_ens_members_min
+                    for i in range(n_ens_members_min)
+                ]
+                if n_model_members == n_ens_members_min:
+                    R_models = np.repeat(R_models, repeats, axis=0)
+                    mu_models = np.repeat(mu_models, repeats, axis=0)
+                    sigma_models = np.repeat(sigma_models, repeats, axis=0)
+                    V_models = np.repeat(V_models, repeats, axis=0)
+                    # For the prob. matching
+                    R_models_pm = np.repeat(R_models_pm, repeats, axis=0)
+
+        R_c = [
+            [R_c[j].copy() for j in range(n_cascade_levels)]
+            for i in range(n_ens_members_max)
+        ]
+        R_c = np.stack(R_c)
+
+        # Check if dimensions are correct
+        assert (
```

Comment:
We prefer to use more explicit if raise blocks, since it allows specifying different errors types that can be catched by users